sourced/ml/algorithms/id_splitter/features.py

```diff
@@ -0,0 +1,125 @@
+    :param identifiers_col: column in CSV file for the raw identifier.
+    :param split_identifiers_col: column in CSV file for the splitted identifier.
+    :param shuffle: indicates whether to reorder the list of identifiers
+        at random after reading it.
+    :return: list of splitted identifiers.
+    """
+    log = logging.getLogger("read_identifiers")
+    log.info("Reading data from CSV...")
+    identifiers = []
+    # TODO: Update dataset loading as soon as https://github.com/src-d/backlog/issues/1212 done.
+    # Think about dataset download step.
+    with tarfile.open(csv_path, mode="r", encoding="utf-8") as f:
+        assert len(f.members) == 1, "One archived file is expected, got: %s" % len(f.members)
+        content = f.extractfile(f.members[0])
+        if not use_header:
+            content.readline()
+        for line in progress_bar(content.readlines(), log):
+            row = line.decode("utf-8").strip().split(",")
+            if len(row[identifiers_col]) <= max_identifier_len:
+                identifiers.append(row[split_identifiers_col])
+    if shuffle:
+        np.random.shuffle(identifiers)
+    log.info("Number of identifiers after filtering: %s." % len(identifiers))
+    return identifiers
+
+
+def prepare_features(csv_path: str, use_header: bool=True,
+                     max_identifier_len: int=DEFAULT_MAX_IDENTIFIER_LEN,
+                     identifiers_col: int=CSV_IDENTIFIERS_COL,
+                     split_identifiers_col: int=CSV_SPLIT_IDENTIFIERS_COL,
+                     shuffle: bool=True, test_size: float=0.2, padding: str=PADDING):
+    """
+    Prepare the features for training the identifier splitting task.
+    :param csv_path: path to the CSV file.
+    :param use_header: uses header as normal line (True) or treat as header line with column names.
+    :param max_identifier_len: maximum length of raw identifier. Skip identifier if longer.
+    :param identifiers_col: column in CSV file for the raw identifier.
+    :param split_identifiers_col: column in CSV file for the splitted identifier.
+    :param shuffle: indicates whether to reorder the list of identifiers at random after reading it
+    :param padding: position where to add padding values:
+        after the intput sequence if "post", before if "pre".
+    :return: training and testing features to train the neural net for the splitting task.
+    """
+    log = logging.getLogger("prepare_features")
+
+    # read data from file
+    identifiers = read_identifiers(csv_path=csv_path, use_header=use_header,
+                                   max_identifier_len=max_identifier_len,
+                                   identifiers_col=identifiers_col,
+                                   split_identifiers_col=split_identifiers_col, shuffle=shuffle)
+
+    # convert identifiers into character indices and labels
+    log.info("Converting identifiers to character indices...")
```

Comment:
Can you please add the number of identifiers and probably other useful stats to this log